src/gui/folderman.cpp

```diff
@@ -95,21 +95,14 @@ FolderMan::~FolderMan()
     _instance = nullptr;
 }
 
-const OCC::Folder::Map &FolderMan::map() const
+const QMap<QString, Folder *> &FolderMan::map() const
 {
     return _folderMap;
 }
 
-QList<Folder *> FolderMan::list() const
-{
-    return _folderMap.values();
-}
-
 void FolderMan::unloadFolder(Folder *f)
 {
-    if (!f) {
-        return;
-    }
+    Q_ASSERT(f);
```

Comment:
This is less defensive than the old code. 
The new one asserts in debug builds which is good, but it could crash *somewhere* in production

Reply:
As I don't use the old unload anymore this should be fixed.